extensions/amp-story/1.0/amp-story.js

```diff
@@ -1589,8 +1581,39 @@ export class AmpStory extends AMP.BaseElement {
   onBookendStateUpdate_(isActive) {
     this.toggleElementsOnBookend_(/* display */ isActive);
     this.element.classList.toggle('i-amphtml-story-bookend-active', isActive);
+    this.updateHistoryState_(HistoryStates.BOOKEND_ACTIVE, isActive);
+  }
+
+  /**
+   * Updates the value for a given state in the window history.
+   * @param {string} stateName
+   * @param {string|boolean} newValue
+   */
+  updateHistoryState_(stateName, newValue) {
```

Comment:
Nit: `setHistoryState_(stateName, value)` so we have a super simple `get/set` API. We could be setting a history state for the first time.